lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -6243,63 +6243,66 @@ namespace Js
             {
                 if (typeHandler->IsSharable())
                 {
+                    bool cachedProtoCanBeCached = nullptr;
+                    if (type->GetPrototype() == JavascriptOperators::GetPrototypeObjectForConstructorCache(constructor, requestContext, cachedProtoCanBeCached)
```

Comment:
Should we check if we have `overridingNewTarget` in JavascriptFunction::CallAsConstructor instead?